Use radio buttons instead of drop down for group #15

vicc30 opened the pull request:
Change to radio buttons Member group field using class --minimal and unorder listing

vicc30 committed:
Use radio buttons instead of drop down for group #15

Change to radio buttons Member group field using class --minimal and unorder listing

vicc30 reviewed (commented):
Hello! I have added the change. Please review it. I have checked it and have worked as it is expected

thewildmage commented on src/vogsphere.html:
```diff
@@ -15,15 +15,33 @@ <h3>Basics</h3>
-                <option value="passenger">Passenger</option>
-            </select>
-        </div>
+        <!--Added radio buttons for member-group selection-->
```
Could you please remove this comment? It's not needed and potentially confusing for people using this tool.

thewildmage commented on src/vogsphere.html:
```diff
@@ -15,15 +15,33 @@ <h3>Basics</h3>
+            <legend>
+                Member Group
+            </legend>
+            <!--Selected class minimal to continue same format below-->
```
While this comment is helpful in the context of this PR, I don't think it's needed long-term, since the CSS is all meant to be replaced. Could you please remove this comment?

thewildmage commented on src/vogsphere.html:
```diff
@@ -15,15 +15,33 @@ <h3>Basics</h3>
+            <!--Selected class minimal to continue same format below-->
+            <ul class="--minimal">
+                <li>
+                    <label for="vogsphere__member-group">
+                        <input type="radio" name="memberGroup" value="crew" id="crew">
```
The `id` has to match what's in the corresponding `for`. Without these matching up, screen readers may announce the wrong thing and also clicking on the label to select the radio button doesn't work. See here for more info: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label

thewildmage commented on src/vogsphere.html:
```diff
@@ -15,15 +15,33 @@ <h3>Basics</h3>
+                    </label>
+                </li>
+                <li>
+                    <label for="vogsphere__member-group">
+                        <input type="radio" name="memberGroup" value="scientist" id="scientist">
```
Same here, `id` should match `for`

thewildmage commented on src/vogsphere.html:
```diff
@@ -15,15 +15,33 @@ <h3>Basics</h3>
+                    </label>
+                </li>
+                <li>
+                    <label for="vogsphere__member-group">
+                        <input type="radio" name="memberGroup" value="passenger" id="passenger">
```
And one more `id` should match `for`

thewildmage commented on src/vogsphere.html:
```diff
@@ -15,15 +15,33 @@ <h3>Basics</h3>
 
-        <div class="__input-group">
-            <label for="vogsphere__member-group">Member Group</label>
-            <select id="vogsphere__member-group" name="memberGroup" required>
```
The `required` attribute that was previously present is now missing. It should be added to all the new `<input>` fields. (See here for more info: https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/required)

vicc30 replied:
Could I change `required` for `checked` tag? with default input "Crew" ?

thewildmage replied:
No, sorry. `required` and `checked` do different things, so they are not interchangeable.

vicc30 replied:
Got it! I'll work on it

thewildmage replied:
Awesome, thanks!

vicc30 replied:
I have added `required` tag to radio selection. But when I was testing it, it is not working the Error message generator so I had to add some JS lines in order to show up that is missing the selection.
`if (!input.memberGroup.value) {`
`            errors.push("ERROR: Missing Member group selection");`
`        }`

Is it ok?

thewildmage reviewed (changes requested):
Hey! I have a few requested changes for you. If you could please take a look at these things, that'd be great. Thanks!

vicc30 commented:
Sure. I'll work on it and thank you for feedback 👍

vicc30 committed:
Reviewed changes

Removed comments. 
Assign correct id for each label for Member Group fieldset.
Added checked tag for input Crew
